### PR TITLE
Better errors

### LINKED ISSUE #150: More informative Dust.js errors
If you have a syntax error with Dust.js it tells you which line and col, but not the file where the error is located in the workspace. Example:

`/workspace/pages/index.dust`

```dust.js
{>"partials/header" /}

{#posts.results}
    {>"partials/post" /}
{/foobar} <<<<<<<<<<<<<<<<<<<<<< Error here

{>"partials/pagination" 
    currentPage=posts.metadata.page
    totalPages=posts.metadata.totalPages
/}

{>"partials/footer" /}
```

```bash
[2017-04-11 11:48:10.409] [LOG]   SyntaxError: Expected end tag for posts.results but it was not found. At line : 5, column : 10
    at Object.compiler.compile (/Users/Dave/Repos/web/node_modules/dustjs-linkedin/lib/compiler.js:34:13)
    at Object.Dust.dust.onLoad (/Users/Dave/Repos/web/dadi/lib/dust/index.js:31:25)
    at /Users/Dave/Repos/web/node_modules/dustjs-linkedin/lib/dust.js:180:18
    at Chunk.map (/Users/Dave/Repos/web/node_modules/dustjs-linkedin/lib/dust.js:722:7)
    at load (/Users/Dave/Repos/web/node_modules/dustjs-linkedin/lib/dust.js:154:22)
    at Object.dust.render (/Users/Dave/Repos/web/node_modules/dustjs-linkedin/lib/dust.js:97:7)
    at Dust.render (/Users/Dave/Repos/web/dadi/lib/dust/index.js:128:8)
    at View.render (/Users/Dave/Repos/web/dadi/lib/view/index.js:36:10)
    at loadData (/Users/Dave/Repos/web/dadi/lib/controller/index.js:206:10)
    at /Users/Dave/Repos/web/dadi/lib/controller/index.js:387:7
    at /Users/Dave/Repos/web/node_modules/async/lib/async.js:52:16
    at Immediate.<anonymous> (/Users/Dave/Repos/web/node_modules/async/lib/async.js:1206:34)
    at runCallback (timers.js:651:20)
    at tryOnImmediate (timers.js:624:5)
    at processImmediate [as _immediateCallback] (timers.js:596:5)
```

### LINKED ISSUE #152: Human friendly errors
Returning JSON isn't friendly to the average human (ie., not devs 🤖)

Slack's error messages for inspiration:

screen_shot_2017-04-02_at_14 38 35_960

### PR DESCRIPTION
**Pull request by abovebored**
Closes #152 
Closes #150

### CODE REVIEW

File: dadi/lib/api/index.js [hunk 1]
```diff
@@ -357,7 +366,16 @@ function notFound (api, req, res) {
       path[0].handler(req, res)
     } else {
       // otherwise, respond with default message
```
Comment (abovebored) [inline dadi/lib/api/index.js]: @jimlambie do we need `res.setHeader('Content-Type', 'text/html')` here as L341?

Comment (jimlambie) [inline dadi/lib/api/index.js]: Yes, I think we do

File: dadi/lib/api/index.js [hunk 2]
```diff
@@ -11,6 +11,8 @@ var url = require('url')
 var log = require('@dadi/logger')
 var config = require(path.join(__dirname, '/../../../config'))
 
+var errorView = require(path.join(__dirname, '/../view/errors'))
```
Comment (jimlambie) [inline dadi/lib/api/index.js]: Do you add the new file to the repo? 😬

Comment (abovebored) [inline dadi/lib/api/index.js]: yeah it's there > /dadi/lib/view/errors.js

**Pull request merged**